nixos/modules/services/backup/restic-rest-server.nix

```diff
@@ -134,6 +134,10 @@ in
     systemd.sockets.restic-rest-server = {
       listenStreams = [ cfg.listenAddress ];
       wantedBy = [ "sockets.target" ];
+      socketConfig = {
+        ReusePort = true;
+        FreeBind = true;
```

Comment:
I don't think we should set this by default. It's good that people see an error message if the interface they're trying to bind to isn't up.

Reply:
Upstream systemd thinks we should, as quoted in the commit message.
I believe we should generally follow such recommendations except for cases where there's a very clear reason not do.

Not setting FreeBind means that you need custom After clauses, depending on your specific interfaces name, just to get a generic server to start reliably.

Note, that even without that option you have no guarantee that the IP is meaningful/routable yet, juste because the interface is up.